tests/create_survey_mt_test.py

```diff
@@ -0,0 +1,130 @@
+#  Copyright (c) 2021 Mira Geoscience Ltd.
+#
+#  This file is part of geoh5py.
+#
+#  geoh5py is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  geoh5py is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+
+# pylint: disable=R0914
+
+import tempfile
+from pathlib import Path
+
+import numpy as np
+import pytest
+
+from geoh5py.objects import Magnetotellurics
+from geoh5py.shared.utils import compare_entities
+from geoh5py.workspace import Workspace
+
+
+def test_create_survey_mt():
+
+    name = "TestMT"
+    n_data = 12
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        path = Path(tempdir) / r"testMT.geoh5"
+
+        # Create a workspace
+        workspace = Workspace(path)
+
+        # Create sources along line
+        x_loc, y_loc = np.meshgrid(np.arange(n_data), np.arange(-1, 3))
+        vertices = np.c_[x_loc.ravel(), y_loc.ravel(), np.zeros_like(x_loc).ravel()]
+
+        # Define the receiver locations on 2 lines, 60 m apart
+        x_loc, y_loc = np.meshgrid(np.linspace(-5, 5, 2), np.linspace(0.0, 20.0, 9))
+        vertices = np.c_[x_loc.ravel(), y_loc.ravel(), np.zeros_like(x_loc).ravel()]
+
+        # Create the survey from vertices
+        mt_survey = Magnetotellurics.create(workspace, vertices=vertices, name=name)
+
+        for key, value in {
+            "input_type": "Rx Only",
+            "survey_type": "Magnetotellurics",
+            "unit": "Hertz (Hz)",
+        }.items():
+            assert getattr(mt_survey, key) == value, f"Error setting defaults for {key}"
+
+        with pytest.raises(TypeError):
```

Comment:
Nice addition!  These can be a little stronger if you check that the output is what you programmed it to be.  Rules out a TypeError popping up in an unexpected place before your code hits:

with pytest.raises(TypeError) as excinfo:
    call_method()

assert "some string from my custom exception" in str(excinfo)

Reply:
Done

Reply:
Sweeeet!